documentation/developer_setup.md

```diff
@@ -124,56 +124,17 @@ import (
 
 var pathToPackage = "sdk/data/aztables/testdata"
 
-type recordingPolicy struct {
-	options recording.RecordingOptions
-	t       *testing.T
-}
-
-func (r recordingPolicy) Host() string {
-	if r.options.UseHTTPS {
-		return "localhost:5001"
-	}
-	return "localhost:5000"
-}
-
-func (r recordingPolicy) Scheme() string {
-	if r.options.UseHTTPS {
-		return "https"
-	}
-	return "http"
-}
-
-func NewRecordingPolicy(t *testing.T, o *recording.RecordingOptions) policy.Policy {
-	if o == nil {
-		o = &recording.RecordingOptions{UseHTTPS: true}
-	}
-	p := &recordingPolicy{options: *o, t: t}
-	return p
-}
-
-func (p *recordingPolicy) Do(req *policy.Request) (resp *http.Response, err error) {
-	if recording.GetRecordMode() != "live" && !recording.IsLiveOnly(p.t) {
-		originalURLHost := req.Raw().URL.Host
-		req.Raw().URL.Scheme = p.Scheme()
-		req.Raw().URL.Host = p.Host()
-		req.Raw().Host = p.Host()
-
-		req.Raw().Header.Set(recording.UpstreamURIHeader, fmt.Sprintf("%v://%v", p.Scheme(), originalURLHost))
-		req.Raw().Header.Set(recording.ModeHeader, recording.GetRecordMode())
-		req.Raw().Header.Set(recording.IDHeader, recording.GetRecordingId(p.t))
-	}
-	return req.Next()
-}
-
 func createClientForRecording(t *testing.T, tableName string, serviceURL string, cred SharedKeyCredential) (*Client, error) {
-	p := NewRecordingPolicy(t, &recording.RecordingOptions{UseHTTPS: true})
-	client, err := recording.GetHTTPClient(t)
+	transport, err := recording.NewRecordingHTTPClient(t)
 	require.NoError(t, err)
 
-	options := &ClientOptions{ClientOptions: azcore.ClientOptions{
-		PerCallPolicies: []policy.Policy{p},
-		Transport:       client,
-	}}
+	options := &ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			Transport:       client,
+		},
+	}
+
+	// Validate the URL ends with a "/"
```

Comment:
We have [`runtime.JoinPaths`](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/runtime/request.go#L51) that will do this for you.